lib/connection.ts

```diff
@@ -533,14 +533,49 @@ export class Connection extends Entity {
    */
   createSession(options?: SessionCreateOptions): Promise<Session> {
     return new Promise((resolve, reject) => {
+      const abortSignal = options && options.abortSignal;
+      let onAbort: Func<void, void>;
+      if (abortSignal) {
+        const rejectOnAbort = () => {
+          const err = createAbortError();
+          log.error("[%s] [%s]", this.id, err.message);
+          return reject(err);
+        };
+
+        onAbort = () => {
+          removeListeners();
+          if (rheaSession.is_open()) {
+            // This scenario *shouldn't* be possible because if `is_open()` returns true,
+            // our `onOpen` handler should have executed and removed this abort listener.
+            // This is a 'just in case' check in case the operation was cancelled sometime
+            // between when the session's state was updated and when the sessionOpen
+            // event was emitted.
```

Comment:
> and when the sessionOpen event was emitted

Nit: Isnt it more like when the sessionOpen event was handled?

Reply:
That is true :smile: